tests/model_fields/test_jsonfield.py

```diff
@@ -494,6 +495,17 @@ def test_nested_key_transform_on_subquery(self):
             [self.objs[4]],
         )
 
+    def test_key_text_transform_char_lookup(self):
+        qs = NullableJSONModel.objects.annotate(
+            char_value=KeyTextTransform("foo", "value"),
+        ).filter(char_value__startswith="bar")
+        self.assertSequenceEqual(qs, [self.objs[7]])
+
+        qs = NullableJSONModel.objects.annotate(
+            char_value=KeyTextTransform(1, KeyTextTransform("bar", "value")),
```

Comment:
I _think_ this one will fail on Postgres at least since `->>` returns text natively.

```suggestion
            char_value=KeyTextTransform(1, KeyTransform("bar", "value")),
```

Reply:
NVM, I forgot that `KeyTransform` aggregated instances of `KeyTransform` independently of their actual types https://github.com/django/django/blob/aed60aee38215e293d6ec2f3c96ec55bb9a62fc2/django/db/models/fields/json.py#L321-L323